addOns/pscanrulesBeta/src/test/java/org/zaproxy/zap/extension/pscanrulesBeta/SubResourceIntegrityAttributeScanRuleUnitTest.java

```diff
@@ -118,52 +120,89 @@ void shouldNotRaiseAlertGivenIntegrityAttributeIsPresentInLinkElement()
+            })
+    void shouldRaiseAlertGivenMissingIntegrityAttributeInLinkElementWithRelevantRel(String relVal)
+            throws HttpMalformedHeaderException {
+        // Given
+        HttpMessage msg =
+                buildMessage(
+                        "<html><head><link rel=\""
+                                + relVal
+                                + "\" href=\"https://site53.example.net/style.css\"></head><body></body></html>");
+        // When
+        scanHttpResponseReceive(msg);
+        // Then
+        assertThat(alertsRaised, hasSize(1));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"dns-prefetch", "DNS-PREFETCH", "DNS-prefetch", "canonical"})
+    void shouldNotRaiseAlertGivenMissingIntegrityAttributeInLinkElementWithIrrelevantRel(
+            String relVal) throws HttpMalformedHeaderException {
+        // Given
+        HttpMessage msg =
+                buildMessage(
+                        "<html><head><link rel=\""
+                                + relVal
+                                + "\" href=\"https://example.net\"></head><body></body></html>");
         // When
         scanHttpResponseReceive(msg);
         // Then
         assertThat(alertsRaised, hasSize(0));
     }
 
     @Test
-    void shouldIndicateElementGivenElementIsServedByTrustedDomainsWhenPatternMismatches()
+    void shouldNotIndicateElementGivenElementIsServedByTrustedDomains()
             throws HttpMalformedHeaderException {
         // Given
         HttpMessage msg =
                 buildMessage(
                         "<html><head>"
                                 + "<script src=\"https://some.cdn.example.com/v1.0/include.js\"></script>"
                                 + "</head><body></body></html>");
-        rule.getConfig()
-                .setProperty(RuleConfigParam.RULE_DOMAINS_TRUSTED, "https://some.cdn.example.com");
+        rule.getConfig().setProperty(RuleConfigParam.RULE_DOMAINS_TRUSTED, "some.cdn.example.com");
         // When
         scanHttpResponseReceive(msg);
         // Then
-        assertThat(alertsRaised, hasSize(1));
+        assertThat(alertsRaised, hasSize(0));
     }
 
     @Test
-    void shouldNotRaiseAlertGivenCanonicalAttributeIsPresentInLinkElement()
+    void shouldIndicateElementGivenElementIsServedByTrustedDomainsWhenPatternMismatches()
             throws HttpMalformedHeaderException {
         // Given
-        // From https://www.w3.org/TR/SRI/#use-casesexamples
         HttpMessage msg =
                 buildMessage(
-                        "<html><head><link href=\"https://dev.example.net/style.css\"\n"
-                                + "rel=\"canonical\"\n"
-                                + "      ></head><body></body></html>");
+                        "<html><head>"
+                                + "<script src=\"https://some.cdn.example.com/v1.0/include.js\"></script>"
+                                + "</head><body></body></html>");
```

Comment:
I figured it would be best to handle all the text block conversion in another PR.

Let me know if you'd like this one to be a block, or if I should tackle them all.

Reply:
I'm fine as is, but I'd ensure that the tests are kept in the same order as they were (maybe it's just the git diff matching making the swap but still).